### DISCUSSION TITLE
Sonoff TX Ultimate - Taking over?

### DISCUSSION THREAD

**Discussion by TeDeVPrime** [Blackymas/NSPanel_HA_Blueprint#2118] [General]
Hello again,
i am trying to use the TX Ultimate switch which in theory i like a lot. (i hope the next version of NSPanel has led as well).
i have flashed it with a custom firmware from here...
https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome

but the project is abandoned and there are a lot of things missing mostly on the led management.

is there any chance that you can show some love to that wall switch as well?

**Comment by edwardtfn** [comment]
I will take a look. 😉

**Reply by TeDeVPrime** [reply]
you beautiful soul. 
i tried to customize but i am hitting one wall after the other. i can't control the leds the way i want them to.
do you have a TX ultimate?

**Reply by edwardtfn** [reply]
I will have one tomorrow. 🙂

**Reply by TeDeVPrime** [reply]
how do i buy you a beer? i already owe you one. 
paypal from the front page?

by the way you will really like the TX Ultimate. so much potential.... but it needs to get proper led notifications which are easy to manipulate.

**Reply by edwardtfn** [reply]
That PayPal goes to Blackymas... I have a link to buy me an ice cream on my GitHub profile.
But don't worry, I do this for fun. 😉

**Reply by TeDeVPrime** [reply]
if it wasn't for you, this project would be dead.
i appreciate your help so here are a few ice creams for you

https://buymeacoffee.com/edwardfirmo/c/9635585

**Comment by plasterer41** [comment]
I would like to buy you a beer also for all your work. You have kept this project alive for many people 🤩Sent from my iPhoneOn 21 May 2024, at 20:06, Edward Firmo ***@***.***> wrote:﻿
That PayPal goes to Blackymas...
But don't worry, I do this for fun. 😉

**Comment by edwardtfn** [comment]
Ready to go... 😉 
![IMG_7074](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/8d59611b-4f5e-48d6-9eb4-593147c9871d)

**Comment by edwardtfn** [comment]
> i have flashed it with a custom firmware from here... https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome
> 
> but the project is abandoned and there are a lot of things missing mostly on the led management

What is the issue with that implementation?

**Reply by TeDeVPrime** [reply]
if you want to start with something easy....

for starters 
1. i want to find a way to set the relays led to toggle at the top and the bottom. right now it only turns on at the bottom.

2. i want to create another "Nightlight" light entity called "AlarmStatus Led" where it turns a chosen side of mine to Red at 50% brightness with pulse effect so that it doesn't stay on all the time.

these are just the first two i need to figure out.

the concept is that i need the TX Ultimate to work as a "distance-notifier" with different led effects.
meaning.... 

first need...
when i go to the bedroom for sleep... i have a TX Ultimate at the entrance of the room.
i need to know if my alarm is armed or not. so i need a selected side (let's say the right side) to be red pulse color... if it is not then i need to arm it.

second need...
i need the TX Ultimate to act a doorbell notification....
so... if someone presses the doorbell... i need a "rainbow" effect for example for 10 seconds

third need...
it will sound crazy so i will tell you later

**Reply by edwardtfn** [reply]
Hahaha
It will certainly be easier if you say the crazy idea earlier than later... 😉

**Reply by TeDeVPrime** [reply]
i am using my TX Ultimate sideways....
the reason for that is that i have window rollers / covers....
so when i slide downwards / right the covers come down... and vice versa.

just above my TX Ultimate i have my NSPanel.

so i want to "JOIN" them as one interface.

this means that i will create an automation where if the shown entity of the NSPanel is a media.... then...
if i slide downwards / right on the TX Ultimate... i want the music volume to go down... and vice versa.
i have the 3 relay switch... 
so if i press the top button (Button 1) i want the previous song
if i press the mid button (Button 2) i want to play/pause
if i press the bottom button (Button 3) i want the next song

if i have a thermostat opened on NSPanel... i want the slide to control temp up or down...

if i have a cover / roller entity on NSPanel... i want to slide the TX Ultimate to control that cover...

so in my mind.... having the wall switch rotated.... makes more sense :)

the problem is that i need a "SWITCH" where i turn the touch relay commands off so that the relays don't work at all when i wish to do other things.

i want to also be able to disable the touch control completely if i want to for security or child reasons but leave the "Touch Hold" command active

in other words....
i wish to convert the TX Ultimate.... as a Ultimate touch controller for the nspanel.
a controller that has a real speaker and it can actually notify with words the user about things...
a controller that has visual notifications with nice led effects....
a perfect companion for the NSPanel.

**Comment by edwardtfn** [comment]
Something is about to come to life: https://github.com/edwardtfn/TX-Ultimate-Easy
🙂

I decided to not fork that other project, but certainly will ~copy~ get a lot of inspiration from there. I will do some effort to enable BLE and this will require significant changes compared to that implementation.
But it still draft, so nothing published yet.

**Reply by TeDeVPrime** [reply]
No problem. I never considered that project to be well-made. it felt rushed.
but... if you can somehow include the features from that project with the one shown in this video...

we might have something good
https://www.youtube.com/watch?v=58v8oqSQgXQ&ab_channel=Unlocoysutecnolog%C3%ADa

he even has the "touch panel power" to stop touch interactions.
and many led effects

**Comment by edwardtfn** [comment]
May I ask you to add your wishes as feature requests on the new repo? 😃 
https://github.com/edwardtfn/TX-Ultimate-Easy
